src/Pos/Delegation/Listeners.hs

```diff
@@ -63,7 +64,9 @@ pskLightRelay = Data $ DataParams $ \pSk -> do
                logDebug $
                    sformat ("Generating delivery proof and propagating it to neighbors: "%build) pSk
                let proof = proxySign SignProxySK sk pSk pSk
-               addToRelayQueue (DataOnlyPM (pSk, proof))
+               -- FIXME seems like a mistake. Shouldn't the relay subsystem
+               -- take care of deiciding when to relay?
```

Comment:
Here we're defining the callbacks for the `DataOnly` relay: what to do when some data comes in. You give it to the relay subsystem, and then it'll take care of the relaying. But this description of when to relay uses the relay subsystem itself. In order to say how to relay, you need to know how to relay. It's just bizarre.